include/triton/Conversion/TritonGPUToLLVM/ElementwiseOpToLLVMBase.cpp

```diff
@@ -1,3 +1,29 @@
+static Type getFunctionType(Type resultType, ValueRange operands) {
+  SmallVector<Type> operandTypes(operands.getTypes());
+  return LLVM::LLVMFunctionType::get(resultType, operandTypes);
+}
+
+static LLVM::LLVMFuncOp
+appendOrGetExternFuncOp(ConversionPatternRewriter &rewriter, Operation *op,
```

Comment:
this is a static function but not used in the cpp. Looking into it, we include this ".cpp" file in the header... We should not do that.
Can we fix this in this PR or in the next one? (cc @zahimoud if he prefers to do it.) Let's make sure it gets addressed soon, that's quite confusing.

Reply:
I defer to @zahimoud on why the include is organized this way. I have made it static precisely for the reason you have mentioned (it is really a .h file in disguise)

Reply:
`static` should never be used on functions defined in a header file, at least not if the header is included in multiple cc files.

This actually does the opposite of what you probably want.  If you have a function defined in a header, you want multiple cc files to be able to use that same function.  By making it `static`, each cc file gets its own private copy of the function.

Functions defined inside headers should almost always be `inline`, which allows the two definitions in two cc files to be commoned into one definition.  (Note that `inline` has nothing to do with the compiler's inliner.)  Templated functions and functions defined inside the body of a class are implicitly inline.

For the same reason, you should never use an anon ns in a header file.

None of this is to say that you need to change something in this PR.

cc @zahimoud

Reply:
@zahimoud is fixing it

Reply:
Yep, sorry for the confusion!

Reply:
Here: https://github.com/openai/triton/pull/3205